culture/review_culture.md

```diff
@@ -0,0 +1,142 @@
+> If you want to go fast, go alone. If you want to go far, go together.
+
+As a community, we value going far over going fast. To go far, we need to stay
+aligned in our values, goals, and practices. One of the ways that we do this is
+through cultivating review culture.
+
+# On Review Culture
+
+Review culture came to us from the practices of software engineers. Large
+codebases are incredibly complex: they contain the ideas and work of hundreds
+of people, manifested in millions of lines of code. They need to remain
+logically consistent, even as they grow and change. A big part of how engineers
+manage is through code review: every single change to the system is reviewed by
+at least one other person before becoming official, or "merging".
+
+Ensuring that all work is reviewed brings several benefits. Most obviously, it
+leads to higher quality work, because the final output includes feedback and
+suggestions from others. Just as importantly, reviews enable the exchange of
+knowledge, context, and perspective within the team. In a good review, both the
+contributor and the reviewer can teach each other about how work gets done on
+the project. Finally, review ensures that every part of the project is
+understood by at least two people.
+
+# The Review Process
+
+## Changes
+The review process orients around _changes_, which are the units of work being
+reviewed.
+
+A change can take many forms, for example:
+- Adding a new guide or piece of documentation
+- Adding a new module to the codebase
+- Modifying an existing guide or module
+
+
+## Participants in the Review Process
+
+### Contributor
+
+The contributor is the person proposing a change. Anyone can decide to be a
+contributor: the only requirements are that you have a change you need to
+propose, and that you agree to be a respectful participant in the review
+process. (When we have a code of conduct--link to it here.)
+
+### Reviewer
+
+The reviewers are the people who review the change, suggest changes to it, and
+eventually approve it. Anyone in the community can act as a reviewer: the only
+requirements are that you have some insight to share (or questions to ask)
+about the change, and that you agree to be a respectful participant in the
+process.
+
+### Maintainer
```

Comment:
Above you say that “every single change to the system is reviewed by at
least one other person before becoming official”. Should it be noted
somewhere that at least one reviewer must be a maintainer? I think that
this is a helpful distinction, because one common point of confusion
from those unfamiliar with free software is, “if anyone can change it,
how doesn’t it just get broken from poor contributions, much less overt
vandalism?” Indicating that at least one reviewer must be explicitly
_trusted_ rather than “anyone in the community” addresses this.

May also want to mention something about this in “Approval and merging”
below.

Reply:
I actually don't think it's a requirement that every approved change was approved by at least one maintainer. Rather, I think every change needs to accumulate "sufficient review" for its level of complexity and potential controversy.

For example, suppose there's a change which fixes a typo in the README. Because the complexity and potential for controversy is minimal, a review by any reasonably-trusted member of the community is sufficient. On the other hand, a massively invasive cross-cutting change may need approval from two maintainers before being approved (in practice this might look like the first maintainer commenting with "this makes sense to me, but because it's so invasive, i'll let (other_maintainer) give canonical approval".

im open to an improvement to the guide that elaborates on this... weak preference to merge the guide as is. (as a general principle, i don't want merging docs to block on every potential improvements--that's what future pull requests are for.)

Reply:
+1, to add a section about it. But also suggest a new PR for this.

(In my experience this point varies wildly between communities. So this document seems like the right place to touch on it.)

Reply:
I think (at the end of the day) someone with write access has to be the one to merge the PR, so a sense the final say always comes down to someone that is technically a maintainer. But a maintainer being pinged to merge a PR that has accumulated sufficient review without reviewing him/herself is different than a maintainer doing the actual review. But if I put myself in those shoes, even if we had a rule about accumulated sufficient review, if I were a maintained pinged to merge it, I'd still want to take a look.

If a tree falls in the woods does it make a sound... if a maintainer browses to a pull request... something something :)